libs/pages/application/src/lib/feature/page-settings-general-feature/page-settings-general-feature.tsx

```diff
@@ -0,0 +1,78 @@
+import { Application, BuildModeEnum, BuildPackLanguageEnum } from 'qovery-typescript-axios'
+import { useEffect } from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
+import { useDispatch, useSelector } from 'react-redux'
+import { useParams } from 'react-router-dom'
+import { editApplication, getApplicationsState } from '@console/domains/application'
+import { ApplicationEntity } from '@console/shared/interfaces'
+import { AppDispatch, RootState } from '@console/store/data'
+import PageSettingsGeneral from '../../ui/page-settings-general/page-settings-general'
+
+export function PageSettingsGeneralFeature() {
+  const { applicationId = '' } = useParams()
+  const dispatch = useDispatch<AppDispatch>()
+  const application = useSelector<RootState, ApplicationEntity | undefined>(
+    (state) => getApplicationsState(state).entities[applicationId]
+  )
+
+  const methods = useForm({
+    mode: 'onChange',
+  })
+
+  const watchBuildMode = methods.watch('build_mode')
+
+  const onSubmit = methods.handleSubmit((data) => {
+    if (data) {
+      const cloneApplication = Object.assign({}, application as Application)
```

Comment:
All this function is untested, I would exteriorize the logic part with the clone and the if else in an outside function that you can test mocking the parameters and asserting the return value.

I would also check that on a submit of the form with act() click for example, a dispatch is well sent